File: contestOfKotlin/src/test/kotlin/challenges/Challenge5Test.kt
package challenges
import org.example.challenges.Challenge5
import kotlin.test.*

class Challenge5Test {
    @Test fun testTriangles() {
        val triangle1 = Challenge5.Triangle(base = 20.0, height = 20.0)
        assertEquals(expected = 200.0, actual = triangle1.getArea(), 0.001)
        assertTrue { triangle1.isEquilateral }
        assertFalse { triangle1.isRectangular }
        assertTrue { triangle1.isIsosceles }

        val triangle2 = Challenge5.Triangle(base = 30.0, height = 30.0, firstAngle = 90.0)
        assertEquals(expected = 450.0, actual = triangle2.getArea())
        assertFalse { triangle2.isEquilateral }
        assertTrue { triangle2.isRectangular }
        assertFalse { triangle2.isIsosceles }

        val triangle3 = Challenge5.Triangle(base = 30.0, height = 20.0, firstAngle = 30.0)
        assertEquals(expected = 300.0, actual = triangle3.getArea())
        assertFalse { triangle3.isEquilateral }
        assertFalse { triangle3.isRectangular }
        assertFalse { triangle3.isIsosceles }
    }

    @Test fun testRectangles() {
        val rectangle1 = Challenge5.Rectangle(length = 30.0, width = 20.0)
        assertEquals(600.0, rectangle1.getArea(), 0.001)
        assertFalse { rectangle1.isEquilateral }
        assertFalse { rectangle1.isSquare }

        val rectangle2: Challenge5.Rectangle = Challenge5.Rectangle(length = 30.0)
        assertEquals(900.0, rectangle2.getArea(), 0.001)
        assertTrue { rectangle2.isEquilateral }
        assertTrue { rectangle2.isSquare }
    }
}

File: contest_of_dart/test/challenges/challenge5_test.dart
import 'dart:math';

import 'package:test/test.dart';
import '../../bin/challenges/challenge5.dart';

void main() {
  test("Challenge 5: Triangles", () {
    final triangle1 = Triangle(20, 20);
    expect(triangle1.getArea(), equals(200));
    expect(triangle1.isEquilateral, equals(true));
    expect(triangle1.isRectangular, equals(false));
    expect(triangle1.isIsosceles, equals(true));

    final triangle2 = Triangle(30, 30, 90);
    expect(triangle2.getArea(), equals(450));
    expect(triangle2.isEquilateral, equals(false));
    expect(triangle2.isIsosceles, equals(false));
    expect(triangle2.isRectangular, equals(true));

    final triangle3 = Triangle(30, 20, 30);
    expect(triangle3.getArea(), equals(300));
    expect(triangle3.isEquilateral, equals(false));
    expect(triangle3.isRectangular, equals(false));
    expect(triangle3.isIsosceles, equals(false));
  });

  test("Challenge 5: Rectangles", () {
    final rectangle1 = Rectangle(30, 20);
    expect(rectangle1.getArea(), equals(600));
    expect(rectangle1.isEquilateral, equals(false));
    expect(rectangle1.isSquare, equals(false));

    final rectangle2 = Rectangle(30, 30);
    expect(rectangle2.getArea(), equals(900));
    expect(rectangle2.isEquilateral, equals(true));
    expect(rectangle2.isSquare, equals(true));
  });
}
